q2_fmt/_peds.py

```diff
@@ -468,3 +484,89 @@ def _create_masking(time_metadata, donor_df, recip_df, reference_column):
+                   drop_incomplete_subjects: bool = False,
+                   drop_incomplete_timepoint: list = None,
+                   bootstrap_replicates: int = 999) -> (pd.DataFrame):
+
+    metadata_df = metadata.to_dataframe()
+    donor = metadata_df[metadata_df.index.isin(
+        _check_reference_column(metadata=metadata_df,
+                                reference_column=reference_column))]
+
+    recipient = metadata_df.loc[metadata_df[reference_column].notnull()]
+    fake_donor = pd.DataFrame([])
+    for i in range(0, bootstrap_replicates+1):
+        if i == 0:
+            peds = sample_peds(
+                table=table, metadata=metadata,
+                time_column=time_column,
+                reference_column=reference_column,
+                subject_column=subject_column,
+                filter_missing_references=filter_missing_references,
+                drop_incomplete_subjects=drop_incomplete_subjects,
+                drop_incomplete_timepoint=drop_incomplete_timepoint)
+            real_temp = peds["measure"]
+        else:
+            shifted_list = recipient[reference_column].sample(frac=1).to_list()
+            recipient.loc[:, reference_column] = shifted_list
+            metadata_df = pd.concat([donor, recipient])
+            metadata = qiime2.Metadata(metadata_df)
+            peds = sample_peds(
+                table=table, metadata=metadata,
+                time_column=time_column,
+                reference_column=reference_column,
+                subject_column=subject_column,
+                filter_missing_references=filter_missing_references,
+                drop_incomplete_subjects=drop_incomplete_subjects,
+                drop_incomplete_timepoint=drop_incomplete_timepoint)
+            fake_donor[i] = peds["measure"]
+    fake_donor_series = fake_donor.median(axis=1)
+    # Common Langauge effect size calcs in prep for stats refactor.
+    # These are not being used currently but will be soon
+    fake_donors_CLES = fake_donor.median(axis=0)
+    agree = ((fake_donors_CLES <
+              real_temp.median()).sum())/bootstrap_replicates
+    disagree = 1 - agree
+    cles = agree - disagree
+    cles
+
+    s, p = mannwhitneyu(real_temp, fake_donor_series.to_list(),
+                        alternative='greater')
+
+    stats_df = pd.DataFrame([["real_values", real_temp.size,
+                              real_temp.median(), "fake_values",
+                              fake_donor_series.size,
+                              fake_donor_series.median(),
+                              fake_donor_series.size,
+                              s, p, np.nan]])
+    stats_df.columns = ['A:group', 'A:n', 'A:measure', 'B:group', 'B:n',
+                        'B:measure', 'n', 'test-statistic', 'p-value',
+                        'q-value']
+
+    stats_df['A:group'].attrs.update({'title': 'real_values',
+                                     'description': '...'})
```

Comment:
Can you update the `...` here (and in the next few lines) to contain descriptions?